src/pipelines/token_classification.rs

```diff
@@ -1129,12 +1130,18 @@ impl TokenClassificationModel {
             }
             LabelAggregationOption::Mode => {
                 let counts = tokens.iter().fold(HashMap::new(), |mut m, c| {
-                    *m.entry((c.label_index, c.label.as_str())).or_insert(0) += 1;
+                    let entry = m
+                        .entry((c.label_index, c.label.as_str()))
+                        .or_insert((0, 0.));
+                    *entry = (
+                        entry.0 + 1,
+                        if c.score > entry.1 { c.score } else { entry.1 },
+                    );
```

Comment:
I think we can use [`max`]() here to clean up the second part of that tuple. And then, so we don't have to remember what `.0` and `.1` are we can maybe destructure the entry at the beginning:
```suggestion
                    let (ref mut count, ref mut score) = m
                        .entry((c.label_index, c.label.as_str()))
                        .or_insert((0, 0.0_f64));
                    *count += 1;
                    *score = score.max(c.score);
```